src/devices/GoPiGo3/GoPiGo3.cs

```diff
@@ -182,20 +180,15 @@ public int SpiRead32(SpiMessageType MessageType)
         /// <returns>Returns the value read</returns>
         public short SpiRead16(SpiMessageType MessageType)
         {
-            short retVal = -1;
             byte[] outArray = { SpiAddress, (byte)MessageType, 0, 0, 0, 0, };
             byte[] reply = SpiTransferArray(outArray);
 
-            if (reply[3] == CorrectData)
+            if (reply[3] == SpiCorrectDataReturned)
             {
-                retVal = (short)((reply[4] << 8) | reply[5]);
+                return BinaryPrimitives.ReadInt16BigEndian(reply.Skip(4).ToArray());
+                // retVal = (short)((reply[4] << 8) | reply[5]);
```

Comment:
BinaryPrimitives seem to work on Span so you can also do `(new Span<byte>(reply)).Slice(4)` - this way you avoid allocating (also please remove commented out code)